### PR TITLE
Feature/library folder export to history

### PR DESCRIPTION
**Pull request by ericenns**
Adds the ability to select a folder for import to a history in the new beta library ui.

### CODE REVIEW

File: client/galaxy/scripts/mvc/library/library-folderrow-view.js [hunk 1]
```diff
@@ -85,7 +85,8 @@ var FolderRowView = Backbone.View.extend({
     tmpl_array.push('  <td>');
     tmpl_array.push('    <span title="Folder" class="fa fa-folder-o"></span>');
     tmpl_array.push('  </td>');
-    tmpl_array.push('  <td></td>');
+    tmpl_array.push('  <td style="text-align: center; "><input style="margin: 0;" type="checkbox"></td>');
+    //tmpl_array.push('  <td></td>');
```
Comment (martenson) [inline client/galaxy/scripts/mvc/library/library-folderrow-view.js]: probably drop this line

File: lib/galaxy/webapps/galaxy/api/history_contents.py [hunk 2]
```diff
@@ -260,6 +270,49 @@ def __create_dataset( self, trans, history, payload, **kwd ):
         return self.hda_serializer.serialize_to_view( hda,
             user=trans.user, trans=trans, **self._parse_serialization_params( kwd, 'detailed' ) )
 
+    def __create_datasets_from_library_folder( self, trans, history, payload, **kwd ):
+        created_hdas = []
```
Comment (martenson) [inline lib/galaxy/webapps/galaxy/api/history_contents.py]: never used yet :)

**Comment by martenson** [comment]
few issues:
- the folder rows cannot be selected by clicking on them (dataset rows can)
- when there are only folder rows present there is no `to History` button (because it shows only when there are datasets present in the view)
- (minor) the folder icon colors are not getting inverted when selected(dataset's are)
- ~~there is no UI confirmation after success (datasets show green mod_toastr message)~~
- ~~(minor) the progress bar doesn't update to 100 if there are only folders being imported~~

**Comment by jmchilton** [comment]
`__create_datasets_from_library_folder` should probably return a dictifiable object that can be dumped to JSON.

File: client/galaxy/scripts/mvc/library/library-foldertoolbar-view.js [hunk 3]
```diff
@@ -289,13 +304,16 @@ var FolderToolbarView = Backbone.View.extend({
...
-            if ( this.parentElement.parentElement.id !== '' ) {
+            if ( this.parentElement.parentElement.id !== '' && this.parentElement.parentElement.classList.contains('dataset_row') ) {
                 dataset_ids.push( this.parentElement.parentElement.id );
+            } else if ( this.parentElement.parentElement.id !== '' && this.parentElement.parentElement.classList.contains('folder_row') ) {
+                folder_ids.push( this.parentElement.parentElement.id );
             }
         } );
     var url = ( window.galaxy_config ? galaxy_config.root : '/' ) + 'api/libraries/datasets/download/' + format;
-    var data = { 'ld_ids' : dataset_ids };
+    var data = { 'ld_ids' : dataset_ids, 'lf_ids' : folder_ids };
```
Comment (martenson) [inline client/galaxy/scripts/mvc/library/library-foldertoolbar-view.js]: plase rename to `folder_ids`

**Comment by martenson** [comment]
Thank you very much @ericenns for the contribution!

**Pull request merged**